packages/Main/src/Parser/PotreeBinParser.js

```diff
@@ -79,30 +106,72 @@ export default {
+        const scale = source.scale;
+        const pointAttributes = source.pointAttributes;
+
+        // find a methode by recursion to get offset from the node id ?
+        const offset = options.in.offsetBBox.min.toArray();
+
+        const forward = (source.crs !== options.in.crs) ?
+            proj4(source.crs, options.in.crs).forward :
+            (x => x);
+        const applyQuaternion = (source.crs !== options.in.crs) ?
+            _applyQuaternion : (x => x);
+
+        const origin = options.in.origin.toArray();
+        const quaternion = options.in.rotation.toArray();
+
         let pointByteSize = 0;
-        for (const potreeName of options.in.pointAttributes) {
+        for (const potreeName of pointAttributes) {
             pointByteSize += POINT_ATTRIBUTES[potreeName].byteSize;
         }
         const numPoints = Math.floor(buffer.byteLength / pointByteSize);
 
         const geometry = new THREE.BufferGeometry();
         let elemOffset = 0;
         let attrOffset = 0;
-        for (const potreeName of options.in.pointAttributes) {
+
+        for (const potreeName of pointAttributes) {
             const attr = POINT_ATTRIBUTES[potreeName];
             const arrayLength = attr.numElements * numPoints;
             const array = new attr.arrayType(arrayLength);
             for (let arrayOffset = 0; arrayOffset < arrayLength; arrayOffset += attr.numElements) {
+                const position = [];
                 for (let elemIdx = 0; elemIdx < attr.numElements; elemIdx++) {
-                    array[arrayOffset + elemIdx] = attr.getValue(view, attrOffset + elemIdx * attr.numByte);
+                    if (attr.attributeName === 'position') {
+                        position.push(attr.getValue(view, attrOffset + elemIdx * attr.numByte)
+                        * scale
+                        + offset[elemIdx]);
+                    } else {
+                        array[arrayOffset + elemIdx] = attr.getValue(view, attrOffset + elemIdx * attr.numByte);
+                    }
+                }
+                if (attr.attributeName === 'position') {
+                    const [x, y, z] = forward(position);
+
+                    const position2 = applyQuaternion([
+                        x - origin[0],
+                        y - origin[1],
+                        z - origin[2],
+                    ], quaternion);
+
+                    array[arrayOffset + 0] = position2[0];
+                    array[arrayOffset + 1] = position2[1];
+                    array[arrayOffset + 2] = position2[2];
                 }
+
                 attrOffset += pointByteSize;
             }
             elemOffset += attr.byteSize;
             attrOffset = elemOffset;
+
             geometry.setAttribute(attr.attributeName, new THREE.BufferAttribute(array, attr.numElements, attr.normalized));
         }
 
+        geometry.userData.origin = options.in.origin;
+        geometry.userData.rotation = options.in.rotation;
```

Comment:
it's mesh transformation ?

Reply:
PR using THREE in LASLoader ?

This is commun at all parser.